### ISSUE TITLE
Phased training doesn't evaluate the model against new knowledge or skills

### ISSUE THREAD

**Issue by danmcp** [instructlab/instructlab#2314]
**Describe the bug**
Phased training uses mt_bench and mmlu but not the _branch variants.  So we aren't validating models have learned the new knowledge and skills.  This makes it likely early models will be picked since they are most likely to answer well on the standard benchmarks vs models trained more on the new knowledge.

**To Reproduce**
Run phased training

**Expected behavior**
Selected model should be some mixture of the best model overall + the best model with my submitted knowledge/skill qnas

**Comment by danmcp** [comment]
@JamesKunstle Not sure if this issue exists elsewhere, curious of your thoughts.

**Comment by JamesKunstle** [comment]
@danmcp You're completely right. One way we're mitigating this problem is to push MMLU checkpoint selection behind the creation of the base-model. MMLU will be used to make sure that the the instruction-tuned model performs well on instruction-based tasks.

We then pick a checkpoint post phase 1 training that, via experimental trends, we are confident will have picked up user knowledge satisfactorily. Right now, that's epoch 7, and we've noted that in the training loop.

Neither of these minor fixes solve the problem you raise, however: we ought to pick a best checkpoint based on user data. That's something to investigate and implement.
